src/main/java/zmaster587/libVulpes/util/FluidUtils.java

```diff
@@ -63,7 +63,7 @@ public static boolean attemptDrainContainerIInv(EmbeddedInventory inv, IFluidHan
 			boolean fill = false;
 			boolean toReturn = false;
 			FluidActionResult modifiedContainer = null;
-			if (!(stack.getItem() == Items.BUCKET) && ((getFluidForItem(stack) != null && getFluidForItem(stack).amount == getFluidItemCapacity(stack)) || tank.getTankProperties()[0].getContents().amount == 0)) {
+			if (!(stack.getItem() == Items.BUCKET) && ((getFluidForItem(stack) != null && getFluidForItem(stack).amount == getFluidItemCapacity(stack)) || tank.getTankProperties()[0].getContents() == null)) {
```

Comment:
I'd add `|| tank.getTankProperties()[0].getContents().amount == 0` at the end of that if clause to ensure you actually have an empty tank.

Reply:
Okay, I will answer you here too :I
Not needed unless mod developer made custom implementation of drain function and it doesn't care about nullifying fluid
![image](https://user-images.githubusercontent.com/26705162/149840833-b9baca7b-b49b-496d-ae67-afdd2339838b.png)